webapi/src/main/java/org/sakaiproject/webapi/controllers/SiteEntityController.java

```diff
@@ -0,0 +1,513 @@
+                    }
+
+                    if (optGroupRefs.isPresent()) {
+                        try {
+                            Set<String> groupRefs = optGroupRefs.get();
+                            if (groupRefs.isEmpty()) {
+                                if (AccessMode.GROUPED.equals(resourceEdit.getAccess())) {
+                                    resourceEdit.clearGroupAccess();
+                                }
+                            } else {
+                                resourceEdit.setGroupAccess(Set.copyOf(groupRefs));
+                            }
+                        } catch (PermissionException e) {
+                            log.error("Pervious permission check was not sufficient: {}",
+                                    ExceptionUtils.getStackTrace(e));
+                            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+                        } catch (InconsistentException e) {
+                            log.error("InconsistentException setting group access to resource: {}",
+                                    ExceptionUtils.getStackTrace(e));
+                        }
+                    }
+
+                    if (updateAvailability || optGroupRefs.isPresent()) {
+                        try {
+                            contentHostingService.commitResource(resourceEdit);
+                        } catch (ServerOverloadException | OverQuotaException e) {
+                            log.error("Could not commit resource edit due to {}: {}", e.toString(),
+                                    ExceptionUtils.getStackTrace(e));
+                            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+                        }
+                    }
+
+                    ContentResource updatedResource;
+                    try {
+                        updatedResource = contentHostingService.getResource(patchEntity.getId());
+                    } catch (PermissionException | IdUnusedException | TypeException e) {
+                        // This should be safe at this point
+                        log.error("Pervious check was not sufficient: {} {}", e.toString(),
+                                ExceptionUtils.getStackTrace(e));
+                        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+                    }
+                    updatedEntities.add(SiteEntityRestBean.of(updatedResource));
+                    break;
+                case FORUM:
+                    DiscussionForum forum = (DiscussionForum) toolEntities.get(entityKey(patchEntity));
+
+                    Optional.ofNullable(patchEntity.getOpenDate())
+                            .ifPresent(openDate -> forum.setOpenDate(Date.from(openDate)));
+
+                    Optional.ofNullable(patchEntity.getCloseDate())
+                            .ifPresent(closeDate -> forum.setCloseDate(Date.from(closeDate)));
```

Comment:
TODO: Dates do update here, but at least the two availability boolean need to be set. It' might also be necessary to create a new forum, as I observed that when setting it the dates through the UI.